### ISSUE TITLE
Pre-computation of neighbours in constant memory. 

### ISSUE THREAD

**Issue by PipGrylls** [dquigley533/GPU_2DIsing#1]
The neighbours array exists in two memory locations, shared and constant.
It looks like the neighbours could also be sent to constant memory only? 
Why do we use this memory location, this looks like it could also go into texture memory to save some of the indexing?
If this program is more memory bound than cpu bound is there benefit to this could we actually run this computation on chip?

**Comment by dquigley533** [comment]
Yep - neighbours were being calculated and stored in two locations, but using either of these failed to generate any speedup. That's the why the does actually *does* compute the neighbour locations on-the-fly rather than looking them up. The neighbour arrays created on the device aren't actually used.
